Automattic/woocommerce-payments
# RPP: Minimal Checkout

Fixes #7497

#### Changes proposed in this Pull Request

This PR adds everything needed for a happy-path basic payment to the new checkout process.

- Empty `SystemErrorState` and `PaymentErrorState` classes are added.
- `PaymentContext` receives support for the following props: `currency`, `manual_capture`, `metadata`, `level3_data`, `metadata`, `cvc_confirmation`, `fingerprint`, `user_id` and `customer_id`
- `PaymentRequest`:
	- Has new getters for `cvc_confirmation` and `fingerprint`.
	- Populates the two new proeprties into the context.
- The `process` method of `InitialState` now calls all needed services to process a simple payment:
	- Import data from the order to the context through the order service.
	- Generate metadata through the order service.
	- Generate level3 data through the new Level3 service.
	- Create a customer (no updates yet).
	- Create an intent.
	- Deletegate to the order service to add all necessary details to the order, and update its status.
- `OrderService` is now responsible for both loading data, and post-processing. We will need to split it in a meaningful way, suggestions are welcome. Keep in mind that we'll probably work on them later through a separate issue.
- The new `PaymentRequestService` is responsible for generating API/server requests, keeping things composed instead of inheriting or huge classes.
- The legacy `WC_Payments_Customer_Service` class has a new method to create a customer based on order details. Should we keep it there, or add it to a new service?

#### Testing instructions

As always, check tests first, and make sure that they are sound.

With this PR, checkout with a saved/new payment method should work, adding the necessary order status, data and notes. Sounds simple, but just try it.

-------------------

- [x] Covered with tests (100% coverage 😎)
